app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationController.scala

```diff
@@ -125,9 +125,14 @@ class BusinessIdentificationController @Inject()(
             .fold(
               formWithErrors => Future successful Ok(html.business_details(formWithErrors, businessTypeIdentifier)),
               knownFacts =>
-                checkBusinessDetailsGivenValidForm(knownFacts).map { resultWithSession =>
-                  val sessionData = (request.session.data ++ resultWithSession.session.data.toSeq) + ("businessType" -> businessTypeIdentifier)
-                  resultWithSession.withSession(sessionData.toSeq: _*)
+                if (Utr.isValid(knownFacts.utr.value)) {
+                  checkBusinessDetailsGivenValidForm(knownFacts).map { resultWithSession =>
+                    val sessionData = (request.session.data ++ resultWithSession.session.data.toSeq) + ("businessType" -> businessTypeIdentifier)
+                    resultWithSession.withSession(sessionData.toSeq: _*)
+                  }
+                } else {
+                  mark("Count-Subscription-NoAgencyFound")
```

Comment:
There's a couple of places that do these two things now, could introduce a function like `redirectToNoAgencyFound` that does the metrics and then redirect